### PR TITLE
Remove uart blocking writes

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: libraries/AP_IOMCU/AP_IOMCU.cpp [hunk 1]
```diff
@@ -65,7 +65,6 @@ void AP_IOMCU::init(void)
 {
     // uart runs at 1.5MBit
     uart.begin(1500*1000, 128, 128);
-    uart.set_blocking_writes(true);
```
Comment (WickedShell) [inline libraries/AP_IOMCU/AP_IOMCU.cpp:68]: Seems like a large behavior change for what IIRC is a very timing sensitive protocol?

**Comment by tridge** [comment]
looks good, but needs testing:
 - [ ] iomcu
 - [ ] hott telem
 - [ ] linux board (eg. Disco)
 - [ ] flight test on stm32 board

**Comment by rmackay9** [comment]
Now surely this one deserves a description above?

**Comment by lucasdemarchi** [comment]
I'm a little confused on the behavior change for Linux. So right now we would default to blocking writes? UARTDevice that backs the UARTDriver doesn't set O_NONBLOCK by default

File: libraries/AP_IOMCU/AP_IOMCU.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/AP_IOMCU/AP_IOMCU.cpp:75]: This isn't a change in behaviour; Hott telemetry (and indeed, everywhere that sets blocking writes) also marks the serial port as "unbuffered".

Weird thing is that Hott telemetry on master retries sending the byte itself even 'though it marks the port as blocking.

**Pull request closed without merging**